component/client/chat/ChatHistory.tsx

```diff
@@ -3,111 +3,62 @@ import {ListGroup} from "react-bootstrap";
 import * as ReactDOM from "react-dom";
 
 import IData from "../service/IData";
-import RoomService from "../service/RoomService";
 import Message from "./Message";
 
 interface IHistoryState {
-  messages: JSX.Element[];
 }
 
 interface IHistoryProps {
-  username: string;
-  url: string;
-  pushMessage: string;
+  messages: IData[];
 }
 
 class ChatHistory extends React.Component<IHistoryProps, IHistoryState> {
-  private roomService: RoomService;
   private historyEnd: HTMLElement;
   private shouldScroll: boolean;
   private lastPosition: number;
 
   constructor(props: any) {
     super(props);
-
-    this.state = {
-      messages: []
-    };
     this.shouldScroll = true;
-
-    this.instantiateRoomService = this.instantiateRoomService.bind(this);
-    this.createMessage = this.createMessage.bind(this);
     this.handleScroll = this.handleScroll.bind(this);
     this.setHistoryEnd = this.setHistoryEnd.bind(this);
   }
 
   public render(): JSX.Element {
+    const listGroupStyle = {
+      marginBottom: "3px",
+      marginTop: "3px"
+    };
+    console.log(this.props.messages);
     return (
       <div style={{overflow: "auto", height: "268px", minHeight: "268px", maxHeight: "268px"}} onScroll={this.handleScroll}>
-        {this.state.messages}
+        {
+          this.props.messages.map((data, index)=>
+            <ListGroup key={index + data.userFrom + data.message} style={listGroupStyle}>
+              <Message username={data.userFrom} message={data.message} />
+            </ListGroup>
+          )
+        }
         <div style={{float: "left", clear: "both"}}
           ref={this.setHistoryEnd} />
       </div>
     );
   }
 
-  public componentWillUnmount(): void {
-    if (this.roomService) this.roomService.close();
-  }
-
-  public componentDidMount(): void {
-    this.instantiateRoomService(this.props);
-  }
-
   public componentWillReceiveProps(nextProps: IHistoryProps): void {
     // History receives a new message. Send the message and flag to scroll
     // to the bottom of the component.
-    if (nextProps.pushMessage) {
-      this.roomService.pushMessage({userFrom: this.props.username, message: nextProps.pushMessage});
-      this.shouldScroll = true;
-    } else if (nextProps.url !== this.props.url) {
-      console.log(`[ INFO ] : ChatHistory switching from ${this.props.url} to ${nextProps.url}`);
-      this.roomService.close();
-      this.instantiateRoomService(nextProps);
-      this.shouldScroll = true;
-    }
+    this.shouldScroll = true;
```

Comment:
`shouldScroll` controls when to scroll to the bottom of the chat to view new messages. If the user scrolls up, this is set to `false` so that the user does not lose their place while reading the history above.